workorders.lic

```diff
@@ -172,48 +187,148 @@ class WorkOrders
+      bput("get #{bone} stack from my #{@bag}", 'You get')
+      case bput('combine', 'You combine', 'The resulting bone stack')
+      when 'The resulting bone stack'
+        stow_hands
+      when 'You combine'
+        case bput("get #{bone} stack from my #{@bag}", 'You get', 'What were')
+        when 'You get'
+          case bput('combine', 'You combine', 'The resulting bone stack')
+          when 'The resulting bone stack', 'You combine'
+            stow_hands
+          end
+        end
+      end
+#    else
+#      order_item(stock-room, bone-stock)
+    end
+    if inside?("first #{bone} stack", @bag)
+      bone_have = bput("count #{bone} stack", 'You count out \d+ pieces').scan(/\d+/).first.to_i
+      echo ("bone_have is #{bone_have}")
+      pause 3
+      bone_recipe = (quantity * volume)
+      bone_need = ((bone_recipe - bone_have) / 10.0).ceil
+      echo("bone need is #{bone_need}")
+      if bone_have <= bone_recipe
+         bone_need.times do
+           order_item(stock_room, bone_stock)
+           next unless left_hand && right_hand
+           bput('combine', 'You combine')
+         end
+      end      
+      bone_need.times do
+        bput("get #{bone} stack from my #{@bag}", 'You get', 'What were')
+        bput("get #{bone} stack from my #{@bag}", 'You get', 'What were')
+        bput('combine', 'You combine', 'The resulting bone stack', 'You must be')
+      end
+#    end
+      stow_hands
+    else
+      bone_need = ((quantity * volume) / 10.0).ceil
+      echo("bone needed = #{bone_need}")
+      pause 3
+      bone_need.times do
+        order_item(stock_room, bone_stock)
+        next unless left_hand && right_hand
+        bput('combine', 'You combine')
+      end
+      bone_need.times do
+        bput("get #{bone} stack from my #{@bag}", 'You get', 'What were')
+        bput("get #{bone} stack from my #{@bag}", 'You get', 'What were')
+        bput('combine', 'You combine', 'The resulting bone stack', 'You must be')
+      end
+    end
+      stow_hands
+  end
+
   def carve_items(info, item, quantity)
     ensure_copper_on_hand(5000, @hometown)
     recipe, items_per_stock, spare_stock, scrap = find_recipe(info, item, quantity)
-
+    echo("Recipe found and money available")
+     
     quantity.times do |count|
-      if (count % items_per_stock).zero?
-        dispose("#{info['stock-name']} stone") if count > 0 && spare_stock
-        if count > 0
-          go_door
-          pause 0.5 until Room.current.id
-        end
-
-        order_item(info['stock-room'], info['stock-number'])
-        fput('tap my deed')
-        stow_hands
-
-        fput("get #{info['stock-name']} rock")
-        if search?(info['polish-full'])
-          fput('get my surface polish')
-          /(\d+)/ =~ bput('count my polish', 'The surface polish has \d+ uses remaining')
-          if Regexp.last_match(1).to_i < 3
-            dispose('polish')
+      if !@use_bone
```

Comment:
If you have code in both the `if` and `else` case, it is preferred to use a construct like `if a then b else c` (rather than `if !a then c else b`).